### PR TITLE
Report locations for RGBLINK errors with conflicting objects

### LINKED ISSUE #572: Report multiple locations for unionized sections and section fragments
Since those are defined in several locations, it might be better to report all these locations when reporting e.g. a section name conflict, instead of just one and then having the error pop up again. Alternatively, we may just be able to print the amount of fragments, so the user will know to grep for them?

### PR DESCRIPTION
**Pull request by Rangi42**
This requires updating the object file format to record the fstack context for sections themselves, not just for patches. That's okay because we already incremented the object revision number to 11 for the upcoming release.

If/when this gets merged, https://github.com/gbdev/rgbds-obj/pull/10 and https://github.com/gbdev/rgbobj/pull/13 should be merged as well.

Fixes #572

### CODE REVIEW

File: test/link/section-union/align-conflict.out [hunk 1]
```diff
@@ -1,4 +1,4 @@
-error: Section "conflicting alignment" is defined with conflicting 4-byte alignment (offset 0) and address $cafe
+error: Section "conflicting alignment" is defined with 4-byte alignment (offset 0) at section-union/align-conflict.asm(7), but with address $cafe at section-union/align-conflict.asm(7)
```
Comment (Rangi42) [inline test/link/section-union/align-conflict.out:1]: These really are the same file; it's assembled once with `-DSECOND` and once without. Likewise for other `section-union/*.asm` tests.

Comment (ISSOtm) [reply test/link/section-union/align-conflict.out:1]: I feel like we should print the object file names so that this is less confusing. But we can also call this an edge enough case not to care.

Comment (Rangi42) [reply test/link/section-union/align-conflict.out:1]: I'd prefer to leave them out: this seems like an edge case created mainly for the sake of testing, and usually the exact filestack+line number will be more useful than just the object file. (And showing both would be too verbose + usually redundant.)

**Review by ISSOtm — APPROVED** [review]
Phew, that was a big one to review. It's annoying how much repetition there is, but eh.
